File: TriCoachAI/Models/Goal.swift
import Foundation

struct Goal: Codable, Equatable, Identifiable {
    var id: UUID = UUID()
    var type: GoalType
    var targetDate: Date
    var priority: GoalPriority = .a
    /// Target time in seconds, when applicable (e.g. "sub-4h marathon").
    var targetTimeSeconds: Int?
    var status: GoalStatus = .active
    var createdAt: Date = .now

    var weeksUntilTarget: Int {
        max(1, Calendar.current.dateComponents([.weekOfYear], from: .now, to: targetDate).weekOfYear ?? type.recommendedWeeks)
    }
}

File: TriCoachAndroid/app/src/main/java/com/tricoach/android/models/Goal.kt
package com.tricoach.android.models

import kotlinx.serialization.Serializable

@Serializable
data class Goal(
    val id: String = "",
    val type: GoalType,
    val targetDate: String,
    val priority: GoalPriority,
    val targetTimeSeconds: Int? = null,
    val status: GoalStatus = GoalStatus.ACTIVE,
    val createdAt: String = "",
)
